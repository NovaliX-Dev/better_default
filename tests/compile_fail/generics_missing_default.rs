#![allow(dead_code)]

use better_default::BetterDefault;

#[derive(BetterDefault)]
enum Enum<T> {
    #[default]
    Variant(T)
}

#[derive(BetterDefault)]
enum Enum2<T> {
    #[default]
    Variant {
        field: T
    }
}

#[derive(BetterDefault)]
struct Struct<T> {
    field: T
}

fn main() {}