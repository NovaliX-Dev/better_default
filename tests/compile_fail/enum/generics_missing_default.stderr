error[E0277]: the trait bound `T: Default` is not satisfied
 --> tests/compile_fail/enum/generics_missing_default.rs:8:13
  |
8 |     Variant(T)
  |             ^ the trait `Default` is not implemented for `T`
  |
help: consider restricting type parameter `T`
  |
6 | enum Enum<T: std::default::Default> {
  |            +++++++++++++++++++++++

error[E0277]: the trait bound `T: Default` is not satisfied
  --> tests/compile_fail/enum/generics_missing_default.rs:15:16
   |
15 |         field: T
   |                ^ the trait `Default` is not implemented for `T`
   |
help: consider restricting type parameter `T`
   |
12 | enum Enum2<T: std::default::Default> {
   |             +++++++++++++++++++++++
